### PR TITLE
fix: update pinpoint regions

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- updated pinpoint region maps to include us-east-2: https://docs.aws.amazon.com/general/latest/gr/pinpoint.html
- adds support to map tests to specific regions instead of forcing to us-west-2

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] migration test added to check that region mapping doesn't affect already created resources

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sobolk — DISMISSED** [review]

**Review by Amplifiyer — COMMENTED** [review]
Can we confirm this is backwards compatible, specially with customers trying to update their pinpoint resource, will it delete and recreate the resource in the us-east-2 region?

**Comment by lazpavel** [comment]
@Amplifiyer, from the standpoint of the pinpoint resource, we cannot update the existing one so a push will still keep the pinpoint resource from us-east-1 even after update, there are problems trying to add notifications channels as we use the mapping logic to build the pinpoint client instead of getting the region from the pinpoint resource which in turn fails with `Resource not found`. Marking it as draft as it requires some design

**Comment by Amplifiyer** [comment]
We should test and get [this](https://github.com/aws-amplify/amplify-cli/pull/9062) merged as well as part of this PR

File: packages/amplify-category-notifications/src/pinpoint-helper.ts [hunk 1]
```diff
@@ -11,20 +11,32 @@ import {
   $TSMeta,
   AmplifyError,
   AmplifyFault,
+  FeatureFlags,
```
Comment (sobolk) [inline packages/amplify-category-notifications/src/pinpoint-helper.ts:14]: CodeQL says it's unused.

**Review by sobolk — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]
LGTM

**Pull request merged**